crates/bevy_gizmos/src/primitives/dim3.rs

```diff
@@ -0,0 +1,822 @@
+        } = self;
+
+        let normal = *rotation * Vec3::Y;
+
+        [-1.0, 1.0].into_iter().for_each(|sign| {
+            draw_circle(
+                gizmos,
+                *radius,
+                *segments,
+                *rotation,
+                *position + sign * *half_height * normal,
+                *color,
+            );
+        });
+
+        draw_cylinder_vertical_lines(
+            gizmos,
+            *radius,
+            *segments,
+            *half_height,
+            *rotation,
+            *position,
+            *color,
+        );
+    }
+}
+
+// capsule 3d
+
+/// Builder for configuring the drawing options of [`Capsule3d`].
+pub struct Capsule3dBuilder<'a, 'w, 's, T: GizmoConfigGroup> {
+    gizmos: &'a mut Gizmos<'w, 's, T>,
+
+    // Radius of the capsule
+    radius: f32,
+    // Half length of the capsule
+    half_length: f32,
+
+    // Center position of the capsule
+    position: Vec3,
+    // Rotation of the capsule
+    //
+    // default orientation is: the capsule is aligned with `Vec3::Y` axis
+    rotation: Quat,
+    // Color of the capsule
+    color: Color,
+
+    // Number of segments used to approximate the capsule geometry
+    segments: usize,
+}
+
+impl<T: GizmoConfigGroup> Capsule3dBuilder<'_, '_, '_, T> {
+    /// Set the number of segments used to approximate the capsule geometry.
+    pub fn segments(mut self, segments: usize) -> Self {
+        self.segments = segments;
+        self
+    }
+}
+
+impl<'w, 's, T: GizmoConfigGroup> GizmoPrimitive3d<Capsule> for Gizmos<'w, 's, T> {
+    type Output<'a> = Capsule3dBuilder<'a, 'w, 's, T> where Self: 'a;
+
+    fn primitive_3d(
+        &mut self,
+        primitive: Capsule,
+        position: Vec3,
+        rotation: Quat,
+        color: Color,
+    ) -> Self::Output<'_> {
+        Capsule3dBuilder {
+            gizmos: self,
+            radius: primitive.radius,
+            half_length: primitive.half_length,
+            position,
+            rotation,
+            color,
+            segments: DEFAULT_NUMBER_SEGMENTS,
+        }
+    }
+}
+
+impl<T: GizmoConfigGroup> Drop for Capsule3dBuilder<'_, '_, '_, T> {
+    fn drop(&mut self) {
+        if !self.gizmos.enabled {
+            return;
+        }
+
+        let Capsule3dBuilder {
+            gizmos,
+            radius,
+            half_length,
+            position,
+            rotation,
+            color,
+            segments,
+        } = self;
+
+        let normal = *rotation * Vec3::Y;
+
+        [1.0, -1.0].into_iter().for_each(|sign| {
+            // use "-" here since rotation is ccw and otherwise the caps would face the wrong way
+            // around
+            let center = *position + sign * *half_length * normal;
+            let top = center + sign * *radius * normal;
+            draw_cap(gizmos, *radius, *segments, *rotation, center, top, *color);
+            draw_circle(gizmos, *radius, *segments, *rotation, center, *color);
+        });
```

Comment:
Does the comment refer to using a negative `sign` for the bottom cap? The actual code doesn't have "-" in the calculations. If so, maybe just say
```suggestion
        [1.0, -1.0].into_iter().for_each(|sign| {
            // use a negative `sign` for the bottom cap since rotation is ccw and otherwise the cap would face
            // the wrong way around
            let center = *position + sign * *half_length * normal;
            let top = center + sign * *radius * normal;
            draw_cap(gizmos, *radius, *segments, *rotation, center, top, *color);
            draw_circle(gizmos, *radius, *segments, *rotation, center, *color);
        });
```
The comment says "caps" though so it's probably referring to something else?  Or maybe it's just outdated

Reply:
This was superseded by some other cleanup. Thanks for spotting it!